server/task/task-memory/src/test/java/org/apache/james/task/eventsourcing/TaskExecutionDetailsProjectionContract.java

```diff
@@ -56,7 +59,11 @@ default void readDetailsShouldBeAbleToRetrieveASavedRecordWithAdditionalInformat
         testee.update(TASK_EXECUTION_DETAILS_WITH_ADDITIONAL_INFORMATION());
 
         Optional<TaskExecutionDetails> taskExecutionDetails = OptionConverters.toJava(testee.load(TASK_ID()));
-        assertThat(taskExecutionDetails).contains(TASK_EXECUTION_DETAILS_WITH_ADDITIONAL_INFORMATION());
+        assertThat(taskExecutionDetails.get())
+            .isEqualToIgnoringGivenFields(TASK_EXECUTION_DETAILS_WITH_ADDITIONAL_INFORMATION(), "additionalInformation");
+        assertThat(taskExecutionDetails.flatMap(TaskExecutionDetails::getAdditionalInformation).get())
+            .isInstanceOf(MemoryReferenceWithCounterTask.AdditionalInformation.class)
+            .isEqualToComparingOnlyGivenFields(ADDITIONAL_INFORMATION().get(), "getCount");
```

Comment:
why do you need to select a field?

Reply:
there is no equals defined on my test additional information.
Would you prefer if i get the additionalInformation, cast it to the good type and then compare the `value` field of each ?

Reply:
isEqualToComparingFieldByField should be enough then?

Reply:
I simplified the assertion